### ISSUE TITLE
Experienceing graph breaks when compiling e3nn norm 

### ISSUE THREAD

**Issue by TaufeqRazakh** [TaufeqRazakh/e3nn-norm-pt2#1]
I have this [reproducer](https://github.com/TaufeqRazakh/graphnorm) repository to demonstrate the graph breaks with pt2 in norm function in [e3nn](https://github.com/e3nn/e3nn/blob/main/e3nn/o3/_norm.py) as follows
```
from graphnorm import Norm, assert_no_graph_break

import pytest

import torch


@pytest.mark.parametrize("irreps_in", ["", "5x0e", "1e + 2e + 4x1e + 3x3o"])
@pytest.mark.parametrize("squared", [True, False])
def test_norm_no_graph_break(irreps_in, squared) -> None:
    """Check whether norm compiles without graph breaks"""

    mod = Norm(irreps_in, squared=squared)
    x = torch.randn(mod.irreps_in.dim)
    torch._logging.set_logs(graph_breaks=True, bytecode=True, recompiles=True, graph=True)
    assert_no_graph_break(mod,x)
```
I am also attaching the logs, which have all the outputs here.
[norm_repro_test.txt](https://github.com/user-attachments/files/16289805/norm_repro_test.txt)

Requirements for this repro are

- sympy
- scipy
- torch
- opt_einsum_fx

**Comment by mitkotak** [comment]
Hey sorry to barge in here. Had a high level question regarding how the PT2 pipeline looks for you guys. Are you planning to use torch.export ? If yes, are you gonna need backprop support during inference ?

**Comment by TaufeqRazakh** [comment]
CC'ed: @KenichiNomura

I am not sure whether any of the modules require backprop during inference
in Allegro/ downstream to e3nn. Prof. Ken-ichi might know this better. But
I believe we will use newer torch.export() on the PT2 trained model.

-Taufeq

**Comment by mitkotak** [comment]
To be specific from what I understand, Allegro computes forces by differentiating the energies unless that assumption was relaxed.

**Comment by mitkotak** [comment]
@TaufeqRazakh If you are curious, here's what the FX graphs look like in the [new TP2](https://colab.research.google.com/drive/14-f-D5yoIL6pRwnF0KQ1klGKluz66h41?usp=sharing). Have'nt analyzed them in detail but one compiler-savy-way of doing optimizations is by doing graph rewrites as shown [here](https://colab.research.google.com/drive/1FTeYO6sf1Vco8dn0qyWze8WIVWLiXQ3z?usp=sharing#scrollTo=cA6Y990eO4me)

**Comment by mitkotak** [comment]
Update: As I was playing with the code, I realized that we can actually get the old API to compile without graph breaks by setting 
```
import e3nn
e3nn.set_optimization_defaults(jit_script_fx=False)
```

Allegro/Nequip do have graph breaks but we should be good for tiny-nequip. We can even attempt doing an tiny-allegro by porting [this](https://github.com/mariogeiger/allegro-jax/blob/main/allegro_jax/allegro.py)

**Comment by KenichiNomura** [comment]
tiny-allegro sounds great! directly relevant to Taufeq's SC paper.

**Comment by TaufeqRazakh** [comment]
@mitkotak @KenichiNomura Yeah, porting tiny-allegro and ensuring the graph mode is correct like the eager mode sounds right to me!.
